### ISSUE TITLE
Even this module's own example code cannot be applied

### ISSUE THREAD

**Issue by jsf9k** [cisagov/molecule-iam-user-tf-module#2]
# 🐛 Bug Report

I am attempting to use this module in [cisagov/ansible-role-kali](https://github.com/cisagov/ansible-role-kali/blob/improvement/add-dns-profile-randomizer/terraform/user.tf) without success.  I have also attempted to apply [this module's example code](https://github.com/cisagov/molecule-iam-user-tf-module/tree/improvement/upstream-updates-testing/examples/basic_usage), but that fails in the same way.  (To see the code I have been using for this, look [here](https://github.com/cisagov/molecule-iam-user-tf-module/tree/bugfix/remove-providers).)  In either case, I get output that looks like:

```console
Error: Error creating IAM policy ParameterStoreReadOnly-test-ansible-role-kali: AccessDenied: User: arn:aws:iam::<users-account-id>:user/jeremy.frasier is not authorized to perform: iam:CreatePolicy on resource: policy ParameterStoreReadOnly-test-ansible-role-kali
        status code: 403, request id: 48b71837-9c6c-46ce-a364-2a11869a9425
  on .terraform/modules/user.parameterstorereadonly_role_production/policy.tf line 21, in resource "aws_iam_policy" "ssm_policy":
  21: resource "aws_iam_policy" "ssm_policy" {

Error: Error creating IAM Role ParameterStoreReadOnly-test-ansible-role-kali: AccessDenied: User: arn:aws:iam::<users-account-id>:user/jeremy.frasier is not authorized to perform: iam:CreateRole on resource: arn:aws:iam::<users-account-id>:role/ParameterStoreReadOnly-test-ansible-role-kali
        status code: 403, request id: c0d00bc2-26f1-4d99-8449-a9fddc4f110c
  on .terraform/modules/user.parameterstorereadonly_role_production/role.tf line 25, in resource "aws_iam_role" "ssm_role":
  25: resource "aws_iam_role" "ssm_role" {

Error: Error creating IAM policy ParameterStoreReadOnly-test-ansible-role-kali: AccessDenied: User: arn:aws:iam::<users-account-id>:user/jeremy.frasier is not authorized to perform: iam:CreatePolicy on resource: policy ParameterStoreReadOnly-test-ansible-role-kali
        status code: 403, request id: 609a703e-acb2-4c75-9c64-25336d6fbaf4
  on .terraform/modules/user.parameterstorereadonly_role_staging/policy.tf line 21, in resource "aws_iam_policy" "ssm_policy":
  21: resource "aws_iam_policy" "ssm_policy" {

Error: Error creating IAM Role ParameterStoreReadOnly-test-ansible-role-kali: AccessDenied: User: arn:aws:iam::<users-account-id>:user/jeremy.frasier is not authorized to perform: iam:CreateRole on resource: arn:aws:iam::<users-account-id>:role/ParameterStoreReadOnly-test-ansible-role-kali
        status code: 403, request id: 4d25e1ab-cf21-4dfc-b9ac-974eb0f04202

  on .terraform/modules/user.parameterstorereadonly_role_staging/role.tf line 25, in resource "aws_iam_role" "ssm_role":
  25: resource "aws_iam_role" "ssm_role" {
```

*Not only is Terraform attempting to create the roles and policies from [cisagov/ssm-read-role-tf-module](https://github.com/cisagov/ssm-read-role-tf-module) with the default provider, it is also attempting to create these resources in the users account.*  It seems that the wrong provider is somehow being passed down to [cisagov/ssm-read-role-tf-module](https://github.com/cisagov/ssm-read-role-tf-module).

I thought the redefinition of the default provider in [cisagov/ssm-read-role-tf-module](https://github.com/cisagov/ssm-read-role-tf-module) might have something to do with this, so I created [this branch](https://github.com/cisagov/ssm-read-role-tf-module/tree/bugfix/remove-provider-definition) and modified a local branch of this module to use it.  That did not seem to change anything.

## Expected behavior

I would expect the [cisagov/ansible-role-kali](https://github.com/cisagov/ansible-role-kali/blob/improvement/add-dns-profile-randomizer/terraform/user.tf) code to apply without errors.  Failing that, I would definitely expect [this module's example code](https://github.com/cisagov/molecule-iam-user-tf-module/tree/improvement/upstream-updates-testing/examples/basic_usage) to apply without errors.

**Comment by jsf9k** [comment]
I also read [this](https://www.terraform.io/docs/configuration/modules.html#providers-within-modules) in the Terraform documentation, which is what led me to think that perhaps defining providers in the modules is not warranted.

Just mentioning it in case it proves helpful.

**Comment by jsf9k** [comment]
Resolved in #3.
